bioio_ome_zarr/reader.py

```diff
@@ -140,127 +151,212 @@ def _read_immediate(self) -> xr.DataArray:
-            spatial_coeffs[dimensions.DimensionNames.SpatialX],
-        )
-
-    @property
-    def channel_names(self) -> Optional[List[str]]:
-        if self._channel_names is None:
-            try:
-                if "omero" in self._zarr.root_attrs:
-                    self._channel_names = [
-                        str(channel["label"])
-                        for channel in self._zarr.root_attrs["omero"]["channels"]
-                    ]
-            except KeyError:
-                pass
-        return self._channel_names
-
     @staticmethod
     def _get_coords(
         dims: List[str],
         shape: Tuple[int, ...],
         scene: str,
         channel_names: Optional[List[str]],
     ) -> Dict[str, Any]:
-        coords: Dict[str, Any] = {}
+        """
+        Construct coordinate mappings for each dimension, currently only Channel.
+
+        Parameters
+        ----------
+        dims : list of str
+            The dimension names in order (e.g. ["T","C","Z","Y","X"]).
+        shape : tuple of int
+            The lengths of each dimension in `dims`.
+        scene : str
+            Identifier for the current scene, used in default channel IDs.
+        channel_names : list of str or None
+            If provided, use these names for the Channel coordinate; otherwise
+            generate default OME channel IDs.
 
-        # Use dims for coord determination
+        Returns
+        -------
+        coords : dict
+            A mapping from dimension name to coordinate values. Only includes
+            entries for Channel if present in `dims`.
+        """
+        coords = {}
         if dimensions.DimensionNames.Channel in dims:
-            # Generate channel names if no existing channel names
             if channel_names is None:
                 coords[dimensions.DimensionNames.Channel] = [
-                    metadata_utils.generate_ome_channel_id(image_id=scene, channel_id=i)
+                    metadata_utils.generate_ome_channel_id(scene, i)
                     for i in range(shape[dims.index(dimensions.DimensionNames.Channel)])
                 ]
             else:
                 coords[dimensions.DimensionNames.Channel] = channel_names
-
         return coords
 
+    def _get_scale_array(self, dims: Tuple[str, ...]) -> List[float]:
+        """
+        Compute combined scale factors for each dimension by merging the
+        overall and per-dataset coordinate transformations.
+
+        Parameters
+        ----------
+        dims : tuple of str
+            The dimension names in order (e.g. ("T","C","Z","Y","X")).
+
+        Returns
+        -------
+        scale : list of float
+            The elementwise product of the global and dataset-specific scales.
+        """
+        ms = self._multiscales_metadata[self._current_scene_index]
+        overall_scale = ms.get(
+            "coordinateTransformations", [{"scale": [1.0] * len(dims)}]
```

Comment:
I feel like we should perhaps start adding debug or warning level logs for when we assign metadata defaults like this. Ex.
```
dataset_scale = ms["datasets"][self._current_resolution_level]["coordinateTransformations"][0]["scale"]
multiscales = self._multiscales_metadata[self._current_scene_index]
if "coordinateTransformations" not in multiscales:
     log.debug("Assigning default overall scale of '1.0' to each dimension since not found in metadata")
     return dataset_scale
overall_scale = multiscales["coordinateTransformations"][0]["scale"]
return [o * d for o, d in zip(overall_scale, dataset_scale)]
```

Reply:
added in [57205d5](https://github.com/bioio-devs/bioio-ome-zarr/pull/36/commits/57205d5a8bc2327f812e6be8fbe00ab083b6b6c5)

Reply:
Warning level seems appropriate.